### PR TITLE
Actuator updates

### PR DESCRIPTION
**Pull request by courtin**
Updates to the actuator disk propeller and electric motor models for integration into the solar model.

### CODE REVIEW

**Comment by courtin** [comment]
@galbramc these test failures seem like mosek issues on some of the VMs

**Comment by bqpd** [comment]
They sure do! Weird, I thought we just fixed those last month.

File: gpkitmodels/GP/aircraft/motor/motor.py [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    def setup(self, static, state):
+        exec parse_variables(MotorPerf.__doc__)
+
+        return [Pshaft == Q*omega,
+                Pelec == v*i,
+                etam == Pshaft/Pelec,
+                static.Qmax >= Q,
+                v <= static.V_max,
+                TCS([i >= Q*Kv+i0]),
+                TCS([v >= omega/Kv + i*R])
```
Comment (mjburton11) [inline gpkitmodels/GP/aircraft/motor/motor.py]: Are you worried about these not being tight?

File: gpkitmodels/GP/aircraft/motor/motor.py [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+               ]
+
+class Motor(Model):
+    """ Electric Motor Model
+
+    Variables
+    ---------
+    Qstar       .5           [kg/(N*m)]         motor specific torque
```
Comment (mjburton11) [inline gpkitmodels/GP/aircraft/motor/motor.py]: alignment :)

File: gpkitmodels/GP/aircraft/motor/motor.py [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+        constraints = [W >= Qstar*Qmax*g]
+
+        return constraints
+
+class PropulsorPerf(Model):
+    """Propulsor Performance Model
+
+    Variables
+    ---------
+    V_static       1      [cm/s]           inflow velocity for static thrust
```
Comment (mjburton11) [inline gpkitmodels/GP/aircraft/motor/motor.py]: Is this the right way to do static? Should we ask Mark about this

File: gpkitmodels/GP/aircraft/motor/motor_test.py [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+        #pp.substitutions[pp.prop.AR_b] = 15
+        self.cost = pp.motor.Pelec/(1000*units('W')) + p.W/(1000*units('lbf'))
+
+        return fs,p,pp
+
+
+def actuator_propulsor_test():
+
+    test = Actuator_Propulsor_Test()
+    #sol = test.debug()
```
Comment (mjburton11) [inline gpkitmodels/GP/aircraft/motor/motor_test.py]: Can you delete all of these commented lines?

File: gpkitmodels/GP/aircraft/motor/motor_test.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+        self.cost = 1./mp.etam
+        return self.mp, fs
+
+def motor_test():
+    test = Motor_P_Test()
+    sol = test.solve()
+    #sol = test.debug()
+    #print sol.table()
+
+def test():
```
Comment (mjburton11) [inline gpkitmodels/GP/aircraft/motor/motor_test.py]: are you not going to test the individual motors here? I really like that idea

File: gpkitmodels/GP/aircraft/prop/propeller.py [hunk 6]
```diff
@@ -20,37 +24,61 @@ class Propeller_Performance(Model):
...
+    a          295          [m/s]       Speed of sound at altitude
     """
 
     def helper(self, c):
-        " helper function to avoid signomial constraint "
         return 2. - 1./c[self.etaadd]
 
-    def setup(self, static, state):
-        exec parse_variables(Propeller_Performance.__doc__)
+    def setup(self, static, state = None, rho = None, V = None):
```
Comment (mjburton11) [inline gpkitmodels/GP/aircraft/prop/propeller.py]: Are `rho` and `V` necessary? Is `= None` necessary?

File: gpkitmodels/GP/aircraft/prop/propeller.py [hunk 7]
```diff
@@ -20,37 +24,61 @@ class Propeller_Performance(Model):
...
-                etai*(z1 + z2**0.5/etaadd) <= 2,
-               ], static
+        constraints = [eta <= etav*etai,
+                       Tc >= T/(0.5*rho*V**2*pi*R**2),
+                       z2 >= Tc + 1,
+                       etai*(z1 + z2**0.5/etaadd) <= 2,
+                       lam >= V/(omega*R),
+                       CT >= Tc*lam**2,
+                       CP <= Q*omega/(.5*rho*(omega*R)**3*pi*R**2),
+                       eta >= CT*lam/CP,
```
Comment (mjburton11) [inline gpkitmodels/GP/aircraft/prop/propeller.py]: Does this over constrain `eta`?

File: gpkitmodels/GP/aircraft/prop/arccos_fit.py [hunk 8]
```diff
@@ -0,0 +1,15 @@
+import unittest
+from numpy import log, exp, log10, vstack
+from numpy import arccos,arange
+from gpfit.fit import fit
+from numpy.random import random_sample
+i = arange(0.0001,1,.001)
+j = arccos(exp(-i))
+#P = Vdd**2 + 30*Vdd*exp(-(Vth-0.06*Vdd)/0.039)
+#u = vstack((Vdd,Vth))
```
Comment (mjburton11) [inline gpkitmodels/GP/aircraft/prop/arccos_fit.py]: can we delete these lines?

File: gpkitmodels/GP/aircraft/prop/arccos_fit.py [hunk 9]
```diff
@@ -0,0 +1,15 @@
...
+x = log(i)
+y = log(j)
+K = 1
+
+cstrt, rmsErr = fit(x,y,K,"SMA")
```
Comment (mjburton11) [inline gpkitmodels/GP/aircraft/prop/arccos_fit.py]: Are you using this fit somewhere?

**Comment by mjburton11** [comment]
@courtin looking good.  Mostly minor changes.

File: gpkitmodels/GP/aircraft/motor/motor.py (hunk 1 above)
Comment (courtin) [reply gpkitmodels/GP/aircraft/motor/motor.py]: I was at one point.  It depends on how the model is being used.  Is there a high cost to having that check?

File: gpkitmodels/GP/aircraft/motor/motor.py (hunk 3 above)
Comment (courtin) [reply gpkitmodels/GP/aircraft/motor/motor.py]: I believe it is but it's not actually used so I'm deleting it.

File: gpkitmodels/GP/aircraft/prop/propeller.py (hunk 7 above)
Comment (courtin) [reply gpkitmodels/GP/aircraft/prop/propeller.py]: No; eta is determined by eta_v and eta_i and then used to constrain the ratio of output to input power (CT lam)/(CP)

**Comment by courtin** [comment]
@mjburton11 should be fixed now

**Comment by galbramc** [comment]
sorry I'm working on upgrading the VM's and hashing out some of the hickups.

**Comment by galbramc** [comment]
test this please

**Comment by galbramc** [comment]
test models please

**Comment by galbramc** [comment]
test models please

**Comment by galbramc** [comment]
looks like jenkins is back in business, and all the tests are passing for this PR.

File: gpkitmodels/GP/aircraft/motor/motor.py (hunk 1 above)
Comment (mjburton11) [reply gpkitmodels/GP/aircraft/motor/motor.py]: No, but in general I think we should only have TCS when there's a possible use case where it shouldn't be tight.  Let's leave it for now.

File: gpkitmodels/GP/aircraft/motor/motor_test.py [hunk 10]
```diff
@@ -0,0 +1,97 @@
...
+        self.cost = pp.motor.Pelec/(1000*units('W')) + p.W/(1000*units('lbf'))
+
+        return fs,p,pp
+
+
+def actuator_propulsor_test():
+
+    test = Actuator_Propulsor_Test()
+    sol = test.solve()
+    #print sol.table()
```
Comment (mjburton11) [inline gpkitmodels/GP/aircraft/motor/motor_test.py]: let's delete all of `print sol.table()`.  And just have `test.solve()`

File: gpkitmodels/GP/aircraft/prop/propeller.py (hunk 7 above)
Comment (mjburton11) [reply gpkitmodels/GP/aircraft/prop/propeller.py]: Oh right.  It was over constrained in the other one.

**Comment by courtin** [comment]
done

**Comment by mjburton11** [comment]
test models please

**Comment by courtin** [comment]
test models please

**Comment by courtin** [comment]
@galbramc still something strange going on with the tests for this PR - it looks like one if being marked as failed despite all the tests passing.

**Comment by bqpd** [comment]
You have to go to console output to interpret those...one timed out (at 90 minutes! what kind of processor prioritization is going on on that machine anyway :p) while the other two lost internet access for a bit.

**Comment by bqpd** [comment]
test models please

**Comment by galbramc** [comment]
finally. one of the VMs was busy all day.

**Comment by courtin** [comment]
woo!

**Pull request merged**